src/language/dafnyInstallation.ts

```diff
@@ -254,22 +256,25 @@ export class DafnyInstaller {
     this.writeStatus('Starting Dafny installation');
     try {
       await this.cleanInstallDir();
-      if(os.type() === 'Darwin' && os.arch() !== 'x64') {
+      const archive = await this.downloadArchive(await getDafnyDownloadAddress(this.context), 'Dafny');
+      await this.extractArchive(archive, 'Dafny');
+      await workspace.fs.delete(archive, { useTrash: false });
+      this.writeStatus('Dafny installation completed');
+      return true;
+    } catch(error: unknown) {
+      this.writeStatus('Dafny download failed:');
+      this.writeStatus(`> ${error}`);
+      console.error('dafny download failed, trying to build from source', error);
+      try {
         // Need to build from source and move all files from Binary/ to the out/resource folder
-        this.writeStatus(`Found a non-supported architecture OSX:${os.arch()}. Going to install from source.`);
+        this.writeStatus('Failed to download binary distribution of Dafny. Installing from source instead.');
         return await this.installFromSource();
-      } else {
-        const archive = await this.downloadArchive(await getDafnyDownloadAddress(this.context), 'Dafny');
-        await this.extractArchive(archive, 'Dafny');
-        await workspace.fs.delete(archive, { useTrash: false });
-        this.writeStatus('Dafny installation completed');
-        return true;
+      } catch(error: unknown) {
```

Comment:
The problem is that the install from source is only available for mac os users, because it uses the command `await this.execLog('brew install dotnet-sdk');`
Could you either ensure that non-mac users can't install from source, or add OS detection and install dotnet another way if necessary?

Reply:
Oh, good point. I tested this, but only on macOS. :) I'll add a further check.